### PR TITLE
[Button#557] Update Content

### PR DESCRIPTION
**Pull request by robergro**
**Update**:
- Enum
- Properties
- UseCases
- Rename icon to image

**Split the ViewModel to**:
- One main ViewModel
- A new Button ViewModel
- A new IconButton ViewModel

### CODE REVIEW

File: core/Sources/Components/Button/UseCase/GetBorder/ButtonGetBorderUseCase.swift [hunk 1]
```diff
@@ -8,8 +8,9 @@
 
 import Foundation
 
-// sourcery: AutoMockable
+// sourcery: AutoMockable, AutoMockTest
```
Comment (michael-zimmermann) [inline core/Sources/Components/Button/UseCase/GetBorder/ButtonGetBorderUseCase.swift:11]: what is "AutoMockTest"?

File: core/Sources/Components/Button/UseCase/GetSizes/ButtonGetSizesUseCase.swift [hunk 2]
```diff
@@ -31,9 +34,29 @@ struct ButtonGetSizesUseCase: ButtonGetSizesUseCaseable {
...
+    }
+
+    func execute(
+        size: ButtonSize,
+        type: ButtonType
+    ) -> ButtonSizes {
+        let height: CGFloat
+        switch size {
+        case .small:
+            height = 32
```
Comment (michael-zimmermann) [inline core/Sources/Components/Button/UseCase/GetSizes/ButtonGetSizesUseCase.swift:50]: Could you create constants for the magic numbers? Same above.

Comment (michael-zimmermann) [reply core/Sources/Components/Button/UseCase/GetSizes/ButtonGetSizesUseCase.swift:50]: Is the old `func execute` deprecated?

File: core/Sources/Components/Button/UseCase/GetBorder/ButtonGetBorderUseCase.swift (hunk 1 above)
Comment (robergro) [reply core/Sources/Components/Button/UseCase/GetBorder/ButtonGetBorderUseCase.swift:11]: I made the stencil to generate a class which contains the CallsCount and Equal for testing.
You can check this class for example: ProgressBarMainGetColorsUseCaseableMockTest.

For example, if you want test the number of call for the useCase, you just need to do:
```ProgressBarMainGetColorsUseCaseableMockTest.XCTCallsCount(useCaseMock, 2)```
If you want to test the number of call, the value for all parameters and the return value:
```ProgressBarMainGetColorsUseCaseableMockTest.XCTAssert(
            useCaseMock,
            expectedNumberOfCalls: 1,
            givenIntent: intentMock,
            givenColors: stub.themeMock.colors as? ColorsGeneratedMock,
            givenDims: stub.themeMock.dims as? DimsGeneratedMock,
            expectedReturnValue: stub.colorsMock
        )
```

File: core/Sources/Components/Button/UseCase/GetSizes/ButtonGetSizesUseCase.swift (hunk 2 above)
Comment (robergro) [reply core/Sources/Components/Button/UseCase/GetSizes/ButtonGetSizesUseCase.swift:50]: I added a Constant ;) 

And yes, there is two execute function here and the first one is deprecated (message on the protocol)

**Review by michael-zimmermann — APPROVED** [review]

**Pull request merged**